pldm_utils.cpp

```diff
@@ -0,0 +1,92 @@
+// SPDX-License-Identifier: Apache-2.0
+
+#include "xyz/openbmc_project/Common/error.hpp"
+
+#include <fmt/core.h>
+#include <libpldm/base.h>
+#include <libpldm/pldm.h>
+
+#include <phosphor-logging/elog-errors.hpp>
+#include <phosphor-logging/log.hpp>
+
+namespace openpower::dump::pldm
+{
+using namespace phosphor::logging;
+namespace internal
+{
+std::string getService(sdbusplus::bus::bus& bus, const std::string& path,
+                       const std::string& interface)
+{
+    using namespace phosphor::logging;
+    constexpr auto objectMapperName = "xyz.openbmc_project.ObjectMapper";
+    constexpr auto objectMapperPath = "/xyz/openbmc_project/object_mapper";
+
+    auto method = bus.new_method_call(objectMapperName, objectMapperPath,
+                                      objectMapperName, "GetObject");
+
+    method.append(path);
+    method.append(std::vector<std::string>({interface}));
+
+    std::vector<std::pair<std::string, std::vector<std::string>>> response;
+
+    try
+    {
+        auto reply = bus.call(method);
+        reply.read(response);
+        if (response.empty())
+        {
+            log<level::ERR>(fmt::format("Error in mapper response for getting "
+                                        "service name, PATH({}), INTERFACE({})",
+                                        path, interface)
+                                .c_str());
+            return std::string{};
+        }
+    }
+    catch (const sdbusplus::exception::exception& e)
+    {
+        log<level::ERR>(fmt::format("Error in mapper method call, "
+                                    "errormsg({}), PATH({}), INTERFACE({})",
+                                    e.what(), path, interface)
+                            .c_str());
+        return std::string{};
+    }
+    return response[0].first;
```

Comment:
We should make sure more than one service hosts the same object or not. Looks we are always looking for the first service which we got from the `GetObject` API

Reply:
this is what we are doing in ohter places too, there will not be more than 1 service.

Reply:
We will get two services always one that hosts the actual service and the mapper service. 

I think the case you tried in this app, might work if your request works fine with the first service. 

FYA, If the mapper service comes first or one more service going to host then, we won't see the expected result.